### ISSUE TITLE
HTTPSConnection.__init__() got an unexpected keyword argument '_socks_options'

### ISSUE THREAD

**Issue by FireCat999** [mansuf/requests-doh#3]
When i run this code
```
from requests_doh import DNSOverHTTPSSession

session = DNSOverHTTPSSession(provider='cloudflare')
proxy={
    'http':'socks5://127.0.0.1:25344',
    'https':'socks5://127.0.0.1:25344',
}
r = session.get('https://google.com',proxies=proxy)
if r.status_code==200:
    print(r.content)
else:
    print(r.status_code)
```
I get the following error
```
    r = session.get('https://google.com',proxies=proxy)
  File "/home/zxcqwe123/.local/lib/python3.10/site-packages/requests/sessions.py", line 602, in get
    return self.request("GET", url, **kwargs)
  File "/home/zxcqwe123/.local/lib/python3.10/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/zxcqwe123/.local/lib/python3.10/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/home/zxcqwe123/.local/lib/python3.10/site-packages/requests/adapters.py", line 486, in send
    resp = conn.urlopen(
  File "/home/zxcqwe123/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 769, in urlopen
    conn = self._get_conn(timeout=pool_timeout)
  File "/home/zxcqwe123/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 295, in _get_conn
    return conn or self._new_conn()
  File "/home/zxcqwe123/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 1070, in _new_conn
    return self.ConnectionCls(
TypeError: HTTPSConnection.__init__() got an unexpected keyword argument '_socks_options'
```
My proxies are working, I checked them in usual requests
Please, can you help me?

**Comment by mansuf** [comment]
It's seems the library doesn't support socks proxies because i didn't implement it, sorry about that. I will try fix it later.

For now, you might wanna use default requests until i fixed it.

```python
from requests import Session

session = Session()
proxy={
    'http':'socks5://127.0.0.1:25344',
    'https':'socks5://127.0.0.1:25344',
}
r = session.get('https://google.com',proxies=proxy)
if r.status_code==200:
    print(r.content)
else:
    print(r.status_code)
```

**Comment by FireCat999** [comment]
Thanks for the answer! I'm waiting for updates

**Comment by mansuf** [comment]
Hello, sorry for waiting too long, i have so many things to do in real life. Anyway, the bug has been fixed in #4. You can try install the patched version.

Installation guide:

**Make sure you have git before doing this**
```sh
pip uninstall requests-doh
pip install git+https://github.com/mansuf/requests-doh.git@ad5512eb290e315afa2eaea6233b184ebebc5947
```

I will merge the PR (#4) later, probably in 1 week or maybe more

**Comment by mansuf** [comment]
Bugfix is released in [v0.3.2](https://github.com/mansuf/requests-doh/releases/tag/v0.3.2)
